packages/tanstack-react-query/src/internals/createOptionsProxy.ts

```diff
@@ -46,9 +66,19 @@ export interface DecorateQueryProcedure<TDef extends ResolverDef> {
    * @see https://tanstack.com/query/latest/docs/framework/react/reference/infiniteQueryOptions#infinitequeryoptions
    */
   infiniteQueryOptions: TRPCInfiniteQueryOptions<TDef>;
+
+  /**
+   * Useful for query invalidation
+   *
+   * @see https://tanstack.com/query/latest/docs/framework/react/guides/query-invalidation
+   */
+  getQueryKey: (input?: TDef['input']) => TRPCQueryKey;
 }
 
 export interface DecorateMutationProcedure<TDef extends ResolverDef> {
+  _input: TDef['input'];
+  _output: TDef['output'];
```

Comment:
how are these shown in IDE? are they shown before or after the "real" stuff?

Reply:
Before which isn't preferable. We probably don't want these here at all though, I just added them for easy access while experimenting. There will be a better solution I'm sure